Recruit-Training/Advanced-Recruit-Training/Embedded-Teaser/Training/Core/Src/Training.c

```diff
@@ -2,5 +2,59 @@
 
 void trainingTask(uint8_t* data)
 {
+	validData = 0;
+	
+	if (motorCheck(*data) && motorCheck(*data) == motorCheck(*(data + 1)) && lightsCheck(*(data + 2)) ) // set outputArray if motors data are valid, motors are in sync and lights data is valid
+	{
+		for (int i = 0; i < sizeof outputArray / sizeof *outputArray; i++)
+		{
+			outputArray[i] = *(data + i);
+		}	
+	}
+	else
+	{
+		for (int i = 0; i < sizeof outputArray / sizeof *outputArray; i++)
+		{
+			outputArray[i] = 0;
```

Comment:
If only the lights is valid, the outputArray[2] should match the data[2], similarly, if only the motors are valid, outputArray[0] and outputArray[1] should match data[0] and data[1] respectively